### ISSUE TITLE
Schedule jobs by random node picking

### ISSUE THREAD

**Issue by tilmankamp** [mozilla/snakepit#101]
At the moment jobs tend to "fill up" the first node in db order, before the next node is taken into account.

**Comment by kdavis-mozilla** [comment]
Should you use random or round-robin scheduling? Using random scheduling can still fill up nodes.

**Comment by tilmankamp** [comment]
Round-robin is a bit more complicated to implement, but yes, I'll do it round-robin. :-)

**Comment by kdavis-mozilla** [comment]
Yeah after I said it I realized it implies state, which, generally, makes things harder.

What about scheduling on some type of load average obtained from the servers? Then the servers maintain the state.

**Comment by tilmankamp** [comment]
Even better. Then nodes should be tried in an order that's based on an allocation quotient. The quotient would be the number of available resources divided by the number of allocated resources of a given node.
